osu.Game.Rulesets.Osu/Skinning/OsuLegacySkinTransformer.cs

```diff
@@ -102,6 +102,14 @@ public override Drawable GetDrawableComponent(ISkinComponent component)
                             Scale = new Vector2(0.8f),
                             Spacing = new Vector2(-overlap, 0)
                         };
+
+                case OsuSkinComponents.SpinnerBody:
+                    if (Source.GetTexture("spinner-top") != null)
+                        return new LegacyNewStyleSpinner();
+                    else if (Source.GetTexture("spinner-background") != null)
+                        return new LegacyOldStyleSpinner();
```

Comment:
```suggestion
                    bool hasBackground = Source.GetTexture("spinner-background") != null;

                    if (Source.GetTexture("spinner-top") != null && !hasBackground)
                        return new LegacyNewStyleSpinner();
                    else if (hasBackground)
                        return new LegacyOldStyleSpinner();
```

Before:
![Screenshot_2020-07-31_23-57-15](https://user-images.githubusercontent.com/1329837/89048054-3e285680-d38a-11ea-9337-31037a4ca6df.png)

After:
![Screenshot_2020-08-01_00-00-54](https://user-images.githubusercontent.com/1329837/89048067-4385a100-d38a-11ea-974f-67c333985cbc.png)

Skin: https://drive.google.com/drive/folders/1ck7IcwCJSC0QbQowneyTUhCIxMFqZTJR

Reply:
Yeah, wasn't sure about this. Looks like a better check.

Reply:
For future reference, full conditional is:
`newStyleSpinner = SkinManager.UseNewLayout && (SkinManager.IgnoreBeatmapSkin || TextureManager.Load(@"spinner-circle", SkinSource.Beatmap) == null) && TextureManager.Load(@"spinner-background", SkinSource.Skin) == null;`

So this should check if version > 1 and do another check against the beatmap skin too. But I think this is MVP for now.